crates/environ/src/vmoffsets.rs

```diff
@@ -332,6 +338,23 @@ pub trait PtrSize {
         .unwrap()
     }
 
+    // Offsets within `VMContObj`
+
+    /// Return the offset of `VMContObj::contref`
+    fn vmcontobj_contref(&self) -> u8 {
+        0
+    }
+
+    /// Return the offset of `VMContObj::revision`
+    fn vmcontobj_revision(&self) -> u8 {
+        8
+    }
+
+    /// Return the size of `VMHostArray`.
+    fn size_of_vmcontobj(&self) -> u8 {
+        16
```

Comment:
And then this should be something like


```suggestion
        u8::try_from(align(
            u32::from(self.vmcontobj_revision())
                + u32::try_from(core::mem::size_of::<u64>()).unwrap(),
            u32::from(self.size()),
        ))
        .unwrap()
```